src/main/java/org/jenkinsci/plugins/workflow/flow/FlowDurabilityHint.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2017, CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.jenkinsci.plugins.workflow.flow;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Provides hints about just how hard we should try to protect our workflow from failures of the master.
+ * There is a trade-off between durability and performance, with higher levels carrying much higher overheads to execution.
+ * Storage and persistence of data should try to provide at least the specified level (may offer more).
+ *
+ * <p> Implementation note: all implementations should be immutable - this is only an extension point rather than enum because
```

Comment:
Exciting changes! Was reading through and noticed:
> this is only an extension point rather than enum because

which appears mismatched with the current code, which is an enum.

Reply:
Thanks @dwnusbaum - dangling remnant from a refactor, fixed now.